evalml/tests/component_tests/decomposer_tests/test_stl_decomposer.py

```diff
@@ -0,0 +1,466 @@
+import datetime
+
+import numpy as np
+import pandas as pd
+import pytest
+from sklearn.linear_model import LinearRegression
+from sklearn.preprocessing import PolynomialFeatures
+
+from evalml.pipelines.components import STLDecomposer
+
+
+def test_stl_decomposer_init():
+    delayed_features = STLDecomposer(degree=3, time_index="dates")
+    assert delayed_features.parameters == {
+        "degree": 3,
+        "seasonal_period": 7,
+        "time_index": "dates",
+    }
+
+
+def test_stl_decomposer_init_raises_error_if_degree_not_int():
+
+    with pytest.raises(TypeError, match="Received str"):
+        STLDecomposer(degree="1")
+
+    with pytest.raises(TypeError, match="Received float"):
+        STLDecomposer(degree=3.4)
+
+    STLDecomposer(degree=3.0)
+
+
+def test_stl_decomposer_auto_sets_seasonal_period_to_odd(ts_data):
+    X, _, y = ts_data()
+
+    stl = STLDecomposer(seasonal_period=3)
+    assert stl.seasonal_period == 3
+
+    stl = STLDecomposer(seasonal_period=4)
+    assert stl.seasonal_period == 5
+
+
+def build_test_target(subset_y, seasonal_period, transformer_fit_on_data, to_test):
```

Comment:
Worth it to move to conftest?

Reply:
It is!  This is actually handled in this parity [PR](https://github.com/alteryx/evalml/pull/3768/files) and will shift up to the `test_decomposer.py` class.